File: 01_GOF23/DP_examples/C/02_Structural/Adapter/target.c
#include "target.h"

// Player 实现
static void player_play(IPlayer* self, const char* fileName, MediaType type) {
    if (type == MP3) {
        printf("PlayAudio mp3 : %s\n", fileName);
    }
}

void player_init(Player* self) {
    self->base.play = player_play;
}

File: 01_GOF23/DP_examples/CSharp/02_Structural/Adapter/Target.cs
namespace Structural.Adapter;
public interface IPlayer {
    void Play(string fileName, MediaType type);
}

public class Player : IPlayer {
    public virtual void Play(string fileName, MediaType type) {
        if (type == MediaType.mp3)
            Console.WriteLine("PlayAudio mp3 : " + fileName);
    }
}
